FWCore/Framework/src/EventProcessor.cc

```diff
@@ -638,6 +643,12 @@ namespace edm {
         }
       }
     }
+    // Initialize after the deletion of non-consumed unscheduled
+    // modules to avoid non-consumed non-run modules to keep the
+    // products unnecessarily alive
+    if (not branchesToDeleteEarly_.empty()) {
+      schedule_->initializeEarlyDelete(std::move(branchesToDeleteEarly_), *preg_);
```

Comment:
I'd prefer an explicit line to clear the memory as it makes it easier to understand the intent.

Reply:
Thanks, I'll add.